flink-connectors/flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/RMQSink.java

```diff
@@ -105,7 +153,18 @@ public void invoke(IN value) {
 		try {
 			byte[] msg = schema.serialize(value);
 
-			channel.basicPublish("", queueName, null, msg);
+			if (messageCompute == null) {
+				channel.basicPublish("", queueName, null, msg);
+			} else {
+				String rk = messageCompute.computeRoutingKey(value);
+				String exchange = messageCompute.computeExchange(value);
+				channel.basicPublish((exchange != null) ? exchange : "",
+						(rk != null) ? rk : "",
+						(returnListener != null) && messageCompute.computeMandatory(value),
```

Comment:
I would prefer that an`IllegalStateException` is thrown if the user code returns `true` without a `returnListener`.

Reply:
For the two behavior are OK, if you think that throwing an exception is a better pattern I will change it. And in fact, throwing exception is easier to explain in documentation.

Reply:
I think failing fast is a reasonable behavior. Can you also adapt the javadocs?